dbcon/mysql/ha_in_sub.cpp

```diff
@@ -125,10 +137,9 @@ execplan::ParseTree* InSub::transform()
     delete rhs;
     ReturnedColumn* lhs = fGwip.rcWorkStack.top();
     fGwip.rcWorkStack.pop();
-    delete lhs;
 
     fSub = (Item_subselect*)(fFunc->arguments()[1]);
-    idbassert(fSub && fFunc);
+    idbassert(fSub);
```

Comment:
This is a poor man way to send the related fFunc into the log so plz left it intact or suggest your way to relay this information.

Reply:
This is not the case. `idbassert(fSub && fFunc)` will print the string `"fsub && fFunc"` in the log, rather than the actual values of the arguments. Since we already check for `fFunc` at the start of the function, we don't need to add an assertion check on `fFunc` here again.

Reply:
Agree.